src/os/bluestore/BlueStore.cc

```diff
@@ -12306,6 +12306,10 @@ void BlueStore::_zoned_cleaner_thread() {
 
 void BlueStore::_zoned_clean_zone(uint64_t zone_num) {
   dout(10) << __func__ << " cleaning zone " << zone_num << dendl;
+  // TODO: (1) copy live objects from zone_num to a new zone, (2) issue a RESET
+  // ZONE operation to the device for the corresponding zone.
+  shared_alloc.a->zoned_mark_zone_clean(zone_num);
```

Comment:
IMO you shouldn't mark zone as clean in allocator before changes land to DB, i.e. after transaction commit..

Reply:
Fixed.